src/adapters/rhythmone.js

```diff
@@ -0,0 +1,453 @@
+*/
+
+var bidmanager = require('../bidmanager.js'),
+  bidfactory = require('../bidfactory.js'),
+  utils = require('../utils.js');
+
+import {ajax as ajax} from '../ajax';
+
+//function setupGA() {
+//  /* jshint ignore:start */
+//  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
+//
+//  ga('create', 'UA-63935000-31', { 'name': 'r1hbga' });
+//  ga('r1hbga.send', 'pageview');
+//  /* jshint ignore:end */
+//}
+
+function track(debug, p1, p2, p3) {
+  if(debug === true){
+    //window.ga('r1hbga.send', 'event', p1, p2, p3);
+    console.log('GA: %s %s %s', p1, p2, p3 || '');
+  }
+}
+
+//setupGA();
+
+var w = (typeof window !== "undefined" ? window : {});
+w.trackR1Impression = track;
+  
+module.exports = function(bidManager, global, loader){
+
+  var version = "0.9.0.0",
+    defaultZone = "1r",
+    defaultPath = "mvo",
+    bidfloor = 0,
+    currency = "USD",
+    debug = false,
+    auctionEnded = false,
+    requestCompleted = false;
+    
+  if(typeof global === "undefined")
+    global = window;
+      
+  if(typeof bidManager === "undefined")
+    bidManager = bidmanager;
+    
+  if(typeof loader === "undefined")
+    loader = ajax;
+  
+  var rhythmBidderUtilities = {
+    template: function(){return "<div id=\"{0}_wrapper\"></div><script type=\"text/javascript\">("+rhythmBidderUtilities.process.toString()+")(\"{0}\", {1}, {2}, {3}, \"{4}\", {5}, {6});</scr"+"ipt>";},
+
+    process: function(target, markup, bidResponse, bidfloor, currency, seat, url){
+    
+      function frameWrap(target, markup, url){
+        var f = document.createElement("iframe");
+        f.style.border="0";
+        f.scrolling="no";
+        f.seamless="seamless";
+        f.style.height = bidResponse.h+"px";
+        f.style.width = bidResponse.w+"px";
+        
+        target.style.height = bidResponse.h+"px";
+        target.style.width = bidResponse.w+"px";
+        target.appendChild(f);
+        
+        if(markup){
+          f.contentWindow.document.open();
+          f.contentWindow.document.write("<html><head></head><body style=\"margin:0;padding:0;\">"+markup+"</body></html>");
+          if((/(MSIE|Trident|Edge)/).test(window.navigator.userAgent) === false)
+            f.contentWindow.document.close();
+        }
+        else if(url)
+          f.src = url;
+      }
+    
+      var w = window,
+        pjs = (function(){
+          try{
+            while(w){
+              if(w.$$PREBID_GLOBAL$$ || w === window.top) return w.$$PREBID_GLOBAL$$;
+              w = w.parent;
+            }
+          }
+          catch(ex){}
+          return null;
+        })(),
+        responses = (pjs && pjs.getBidResponses ? pjs.getBidResponses() : {}),
```

Comment:
No, but we should have one. Essentially no adapter should be looking at bids submitted by other partners or otherwise using the public API's designated for publishers. What this means practically is that you do not need to access any methods/properties on `pbjs` with the exception of if your adapter needs to expose a JSONP callback handler function.